### ISSUE TITLE
Help needed: NaN observed in training loss

### ISSUE THREAD

**Issue by junjiequ** [huggingface/nanoVLM#167]
I am trying to run the training script using commit: `tags/v0.2 release`. I am blocked by this issue for days and couldn't make the training process work from my environment. If anyone could provide a 2nd pair of eyes on what I am doing wrong, it would be much appreciated. 

Below are the parameters I could see printed for the training output:

-- VLM Configuration ---
VLMConfig(vit_hidden_dim=768, vit_inter_dim=3072, vit_patch_size=16, vit_img_size=256, vit_n_heads=12, vit_dropout=0.0, vit_n_blocks=12, vit_ln_eps=1e-06, vit_cls_flag=False, vit_model_type='google/siglip2-base-patch16-256', lm_hidden_dim=576, lm_inter_dim=1536, lm_rms_eps=1e-05, lm_re_base=100000, lm_max_position_embeddings=1024, lm_base_vocab_size=49152, extra_token_amount=1, lm_vocab_size=49153, lm_n_heads=9, lm_n_kv_heads=3, lm_dropout=0.0, lm_n_blocks=30, lm_attn_scaling=1.0, lm_max_length=1024, lm_use_tokens=False, lm_tie_weights=True, lm_model_type='HuggingFaceTB/SmolLM2-360M-Instruct', lm_tokenizer='HuggingFaceTB/SmolLM2-360M-Instruct', lm_eos_token_id=0, mp_pixel_shuffle_factor=2, mp_image_token_length=64, vlm_extra_tokens={'image_token': '<|image|>'}, vlm_load_backbone_weights=True, vlm_checkpoint_path='/opt/ml/checkpoints', hf_repo_name=None)

--- Training Configuration ---
TrainConfig(lr_mp=0.001, lr_backbones=5e-05, data_cutoff_idx=1000, val_ratio=0.025, batch_size=4, gradient_accumulation_steps=4, mmstar_batch_size=12, max_grad_norm=1.0, eval_in_epochs=False, eval_interval=250, epochs=5, compile=False, resume_from_vlm_checkpoint=False, train_dataset_path='HuggingFaceM4/the_cauldron', train_dataset_name=('tqa',), test_dataset_path='Lin-Chen/MMStar', wandb_entity='HuggingFace', log_wandb=False)

I have tried to run the training job from SageMaker using different instances types. 
- One is with A10G GPUs, for debugging purposes, the log output above is generated on an instance with A10G gpu.
- Another one is with 8X A100G GPUs with DDP. 

Either approach above, I am always seeing this message: `Epoch 3/5, Train Loss: nan | Time: 110.37s | T/s: 1038.91`.

I understood it because of Train Loss: nan, there is no model checkpoint saved, hence there is no model saved...

Could anyone help where I did wrong please? 

Thanks in advance!

**Comment by lusxvr** [comment]
We pushed quite a few updates since then, could you please try again?
